### PR TITLE
chore: Prefer native click implementation by UiObject2

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
https://github.com/appium/appium-uiautomator2-server/issues/377

### CODE REVIEW

**Comment by dpgraham** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/io/appium/uiautomator2/model/internal/GestureController.java  4
         

Complexity decreasing per file
==============================
+ app/src/main/java/io/appium/uiautomator2/model/internal/CustomUiDevice.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dpgraham/appium-uiautomator2-server/pullRequest?prid=6179926&bid=20170903)

**Comment by dpgraham** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/io/appium/uiautomator2/model/internal/GestureController.java  3
         

Complexity decreasing per file
==============================
+ app/src/main/java/io/appium/uiautomator2/model/internal/CustomUiDevice.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dpgraham/appium-uiautomator2-server/pullRequest?prid=6179926&bid=20170903)

**Comment by dpgraham** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/io/appium/uiautomator2/model/internal/GestureController.java  3
         

Complexity decreasing per file
==============================
+ app/src/main/java/io/appium/uiautomator2/model/internal/CustomUiDevice.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dpgraham/appium-uiautomator2-server/pullRequest?prid=6179926&bid=20170903)

File: app/src/main/java/io/appium/uiautomator2/model/internal/GestureController.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    private synchronized static Class<?> getPointerGestureClass() {
+        if (pointerGestureClass == null) {
+            pointerGestureClass = ReflectionUtils.getClass(POINTER_GESTURE_CLASS);
+        }
+        return pointerGestureClass;
+    }
+
+    private void performGesture(Object... gestures) {
+        @SuppressWarnings("RedundantArrayCreation")
+        Method performGestureMethod = method(wrappedInstance.getClass(), "performGesture",
```
Comment (KazuCocoa) [inline app/src/main/java/io/appium/uiautomator2/model/internal/GestureController.java:49]: 👀

**Review by KazuCocoa — APPROVED** [review]

File: app/src/main/java/io/appium/uiautomator2/model/internal/GestureController.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    private void performGesture(Object... gestures) {
+        Method[] methods = wrappedInstance.getClass().getMethods();
+        for (Method method : methods) {
+            if (!method.getName().equals("performGesture")) {
+                continue;
+            }
+
+            method.setAccessible(true);
+            invoke(method, wrappedInstance, gestures);
+            return;
```
Comment (dpgraham) [inline app/src/main/java/io/appium/uiautomator2/model/internal/GestureController.java:56]: Codacy found an issue: [Avoid using a branching statement as the last in a loop.](https://app.codacy.com/manual/dpgraham/appium-uiautomator2-server/pullRequest?prid=6179926)

**Comment by dpgraham** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- app/src/main/java/io/appium/uiautomator2/model/internal/GestureController.java  4
         

Complexity decreasing per file
==============================
+ app/src/main/java/io/appium/uiautomator2/model/internal/CustomUiDevice.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dpgraham/appium-uiautomator2-server/pullRequest?prid=6179926&bid=20170903)

**Comment by dpgraham** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/io/appium/uiautomator2/model/internal/GestureController.java  5
         

Complexity decreasing per file
==============================
+ app/src/main/java/io/appium/uiautomator2/model/internal/CustomUiDevice.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dpgraham/appium-uiautomator2-server/pullRequest?prid=6179926&bid=20170903)

**Comment by dpgraham** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/io/appium/uiautomator2/model/internal/GestureController.java  4
         

Complexity decreasing per file
==============================
+ app/src/main/java/io/appium/uiautomator2/model/internal/CustomUiDevice.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dpgraham/appium-uiautomator2-server/pullRequest?prid=6179926&bid=20170903)

**Comment by dpgraham** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/io/appium/uiautomator2/model/internal/GestureController.java  4
         

Complexity decreasing per file
==============================
+ app/src/main/java/io/appium/uiautomator2/model/internal/CustomUiDevice.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dpgraham/appium-uiautomator2-server/pullRequest?prid=6179926&bid=20170903)

**Pull request merged**